BeVolt/BSP/Simulator/DataGeneration/simulate.py

```diff
@@ -13,91 +14,146 @@
 
 def generate(battery=None):
     global state, mode
+    # Update battery's state
     if battery is not None:
-        # Update battery's state
         battery.update()
-        # Generate ADC values
-        ADC.generate(state, mode, battery.current)
-    else:
-        # Generate ADC values
-        ADC.generate(state, mode)
+    # Generate ADC values
+    ADC.generate(state, mode, battery)
+    # Generate SPI values
+    SPI.generate(state, mode, battery)
 
 
 def display(battery=None):
     global stdscr
-    if battery is not None:
-        stdscr.addstr(5, 0, battery.__str__())
     # Read ADC values
     adc_values = ADC.read()
-    stdscr.addstr(0, 0, "Simulator")
-    stdscr.addstr(1, 0, "=============================")
+    stdscr.addstr(0, 10, "Battery")
+    stdscr.addstr(1, 0, "==============================")
     stdscr.addstr(2, 0, f"ADC:")
     stdscr.addstr(2, 10, f"Low Precision: {adc_values[0][0]}")
     stdscr.addstr(3, 10, f"High Precision: {adc_values[0][1]}")
     # Read Current values
     stdscr.addstr(4, 0, f"Current:")
     stdscr.addstr(4, 10, f"{adc_values[1]} A ")
+    # Read Module values
+    stdscr.addstr(0, 54, "Modules")
+    stdscr.addstr(1, 40, "====================================")
+    module_values = SPI.read()
+    for i, module in enumerate(module_values):
+        stdscr.addstr(i+2, 37, f"{i+1}")
+        stdscr.addstr(i+2, 40, f"| {'X' if module[0] else ' '} | {module[1]/10000:.4f}V | {module[2]/1000:.3f}°C | {module[3]/1000:.3f}°C |")
     # Read LED values
+    stdscr.addstr(0, 90, "LEDs")
+    stdscr.addstr(1, 80, "=======================")
     lights = Lights.read()
-    for i in range(0, 9):
-        stdscr.addstr(i+2, 40, lights_names[i])
+    curses.init_pair(1, curses.COLOR_RED, curses.COLOR_RED)
```

Comment:
Is this for the LEDs? Is this part of this set of changes, or did this come from another branch? If it came from elsewhere, we should make sure to review that branch's changes before doing anything here.

`git blame` shows that @SijWoo changed this line. Is this part of a finished feature?